sys/include/net/gnrc/gnrc_priority_pktqueue.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     net_gnrc_oriority_pktqueue Priority packet queue
+ * @file
+ * @brief       Wrapper for priority_queue that holds gnrc_pktsnip_t* and is
+ *              aware of it's length.
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo <shuguo.zhuo@inria.fr>
+ * @}
+ */
+
+#ifndef GNRC_PRIORITY_PKTQUEUE_H
+#define GNRC_PRIORITY_PKTQUEUE_H
+
+#include <stdint.h>
+#include <priority_queue.h>
+#include <net/gnrc/pkt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef priority_queue_node_t packet_queue_node_t;
+
+/* TODO: Description */
+typedef struct {
+    priority_queue_t queue;
+    uint32_t length;
+    packet_queue_node_t* buffer;
+    size_t buffer_size;
+} packet_queue_t;
```

Comment:
Well, the `buffer_size` reflects the central buffer size which is needed when initializing the queue, while the length is the queue's length..